tools/gn

```diff
@@ -236,13 +236,20 @@ def to_gn_args(args):
       gn_args['target_cpu'] = args.windows_cpu
     else:
         # Building host artifacts
-        gn_args['target_cpu'] = 'x64'
+        if sys.platform == 'win32':
+            gn_args['target_cpu'] = args.windows_cpu
+        else:
+          gn_args['target_cpu'] = 'x64'
 
     # DBC is not supported anymore.
     if args.interpreter:
       raise Exception('--interpreter is no longer needed on any supported platform.')
-    gn_args['dart_target_arch'] = gn_args['target_cpu']
 
+    if sys.platform == 'win32' and gn_args['target_cpu'] == 'x86':
```

Comment:
The Dart build will treat 'x86' and 'ia32' the same way. See: https://github.com/dart-lang/sdk/blob/main/runtime/BUILD.gn#L113. So I think this can stay as
```
gn_args['dart_target_arch'] = gn_args['target_cpu']
```

Reply:
Yes, I found this gn in Dart source code, but if we don't use "ia32", there will be an compilation error:
```
[94/872] CXX obj/third_party/dart/runtime/vm/libdart_vm_precompiled_runtime_product.dart.obj
FAILED: obj/third_party/dart/runtime/vm/libdart_vm_precompiled_runtime_product.dart.obj
ninja -t msvc -e environment.x86 -- ../../buildtools/windows-x64/clang/bin/clang-cl.exe -m32 /nologo /showIncludes @obj/third_party/dart/runtime/vm/libdart_vm_precompiled_runtime_product.dart.obj.rsp /c ../../third_party/dart/runtime/vm/dart.cc /Foobj/third_party/dart/runtime/vm/libdart_vm_precompiled_runtime_product.dart.obj /Fdobj/third_party/dart/runtime/vm/libdart_vm_precompiled_runtime_product_cc.pdb
../../third_party/dart/runtime/vm/dart.cc(221,36): error: use of undeclared identifier 'AOT_Array_elements_start_offset'; did you mean 'Array_elements_start_offset'?
  COMMON_OFFSETS_LIST(CHECK_FIELD, CHECK_ARRAY, CHECK_SIZEOF,
                                   ^
../../third_party/dart/runtime\vm/compiler/runtime_offsets_extracted.h(5757,47): note: 'Array_elements_start_offset' declared here
static constexpr dart::compiler::target::word Array_elements_start_offset = 12;
                                              ^
../../third_party/dart/runtime/vm/dart.cc(221,36): error: use of undeclared identifier 'AOT_Array_elements_start_offset'; did you mean 'Array_elements_start_offset'?
  COMMON_OFFSETS_LIST(CHECK_FIELD, CHECK_ARRAY, CHECK_SIZEOF,
                                   ^
../../third_party/dart/runtime\vm/compiler/runtime_offsets_extracted.h(5757,47): note: 'Array_elements_start_offset' declared here
static constexpr dart::compiler::target::word Array_elements_start_offset = 12;
                                              ^
../../third_party/dart/runtime/vm/dart.cc(221,36): error: use of undeclared identifier 'AOT_Array_element_size'; did you mean 'Array_element_size'?
  COMMON_OFFSETS_LIST(CHECK_FIELD, CHECK_ARRAY, CHECK_SIZEOF,
                                   ^
../../third_party/dart/runtime\vm/compiler/runtime_offsets_extracted.h(5758,47): note: 'Array_element_size' declared here
static constexpr dart::compiler::target::word Array_element_size = 4;
                                              ^
../../third_party/dart/runtime/vm/dart.cc(221,36): error: use of undeclared identifier 'AOT_Array_element_size'; did you mean 'Array_element_size'?
  COMMON_OFFSETS_LIST(CHECK_FIELD, CHECK_ARRAY, CHECK_SIZEOF,
                                   ^
../../third_party/dart/runtime\vm/compiler/runtime_offsets_extracted.h(5758,47): note: 'Array_element_size' declared here
static constexpr dart::compiler::target::word Array_element_size = 4;
                                              ^
../../third_party/dart/runtime/vm/dart.cc(221,36): error: use of undeclared identifier 'AOT_Code_elements_start_offset'; did you mean 'Code_elements_start_offset'?
  COMMON_OFFSETS_LIST(CHECK_FIELD, CHECK_ARRAY, CHECK_SIZEOF,
                                   ^
../../third_party/dart/runtime\vm/compiler/runtime_offsets_extracted.h(5759,47): note: 'Code_elements_start_offset' declared here
static constexpr dart::compiler::target::word Code_elements_start_offset = 76;
                                              ^
../../third_party/dart/runtime/vm/dart.cc(221,36): error: use of undeclared identifier 'AOT_Code_elements_start_offset'; did you mean 'Code_elements_start_offset'?
  COMMON_OFFSETS_LIST(CHECK_FIELD, CHECK_ARRAY, CHECK_SIZEOF,
                                   ^
../../third_party/dart/runtime\vm/compiler/runtime_offsets_extracted.h(5759,47): note: 'Code_elements_start_offset' declared here
static constexpr dart::compiler::target::word Code_elements_start_offset = 76;
                                              ^
../../third_party/dart/runtime/vm/dart.cc(221,36): error: use of undeclared identifier 'AOT_Code_element_size'; did you mean 'Code_element_size'?
  COMMON_OFFSETS_LIST(CHECK_FIELD, CHECK_ARRAY, CHECK_SIZEOF,
                                   ^
../../third_party/dart/runtime\vm/compiler/runtime_offsets_extracted.h(5760,47): note: 'Code_element_size' declared herestatic constexpr dart::compiler::target::word Code_element_size = 4;
                                              ^
../../third_party/dart/runtime/vm/dart.cc(221,36): error: use of undeclared identifier 'AOT_Code_element_size'; did you mean 'Code_element_size'?
  COMMON_OFFSETS_LIST(CHECK_FIELD, CHECK_ARRAY, CHECK_SIZEOF,
                                   ^
../../third_party/dart/runtime\vm/compiler/runtime_offsets_extracted.h(5760,47): note: 'Code_element_size' declared herestatic constexpr dart::compiler::target::word Code_element_size = 4;
                                              ^
../../third_party/dart/runtime/vm/dart.cc(221,36): error: use of undeclared identifier 'AOT_Context_elements_start_offset'; did you mean 'Context_elements_start_offset'?
  COMMON_OFFSETS_LIST(CHECK_FIELD, CHECK_ARRAY, CHECK_SIZEOF,
                                   ^
../../third_party/dart/runtime\vm/compiler/runtime_offsets_extracted.h(5761,47): note: 'Context_elements_start_offset' declared here
static constexpr dart::compiler::target::word Context_elements_start_offset =
                                              ^
../../third_party/dart/runtime/vm/dart.cc(221,36): error: use of undeclared identifier 'AOT_Context_elements_start_offset'; did you mean 'Context_elements_start_offset'?
  COMMON_OFFSETS_LIST(CHECK_FIELD, CHECK_ARRAY, CHECK_SIZEOF,
                                   ^
../../third_party/dart/runtime\vm/compiler/runtime_offsets_extracted.h(5761,47): note: 'Context_elements_start_offset' declared here
static constexpr dart::compiler::target::word Context_elements_start_offset =
                                              ^
../../third_party/dart/runtime/vm/dart.cc(221,36): error: use of undeclared identifier 'AOT_Context_element_size'; did you mean 'Context_element_size'?
  COMMON_OFFSETS_LIST(CHECK_FIELD, CHECK_ARRAY, CHECK_SIZEOF,
                                   ^
../../third_party/dart/runtime\vm/compiler/runtime_offsets_extracted.h(5763,47): note: 'Context_element_size' declared here
static constexpr dart::compiler::target::word Context_element_size = 4;
                                              ^
../../third_party/dart/runtime/vm/dart.cc(221,36): error: use of undeclared identifier 'AOT_Context_element_size'; did you mean 'Context_element_size'?
  COMMON_OFFSETS_LIST(CHECK_FIELD, CHECK_ARRAY, CHECK_SIZEOF,
                                   ^
../../third_party/dart/runtime\vm/compiler/runtime_offsets_extracted.h(5763,47): note: 'Context_element_size' declared here
static constexpr dart::compiler::target::word Context_element_size = 4;
                                              ^
../../third_party/dart/runtime/vm/dart.cc(221,36): error: use of undeclared identifier 'AOT_ContextScope_elements_start_offset'; did you mean 'ContextScope_elements_start_offset'?
  COMMON_OFFSETS_LIST(CHECK_FIELD, CHECK_ARRAY, CHECK_SIZEOF,
                                   ^
../../third_party/dart/runtime\vm/compiler/runtime_offsets_extracted.h(5765,5): note: 'ContextScope_elements_start_offset' declared here
    ContextScope_elements_start_offset = 12;
    ^
../../third_party/dart/runtime/vm/dart.cc(221,36): error: use of undeclared identifier 'AOT_ContextScope_elements_start_offset'; did you mean 'ContextScope_elements_start_offset'?
  COMMON_OFFSETS_LIST(CHECK_FIELD, CHECK_ARRAY, CHECK_SIZEOF,
                                   ^
../../third_party/dart/runtime\vm/compiler/runtime_offsets_extracted.h(5765,5): note: 'ContextScope_elements_start_offset' declared here
    ContextScope_elements_start_offset = 12;
    ^
../../third_party/dart/runtime/vm/dart.cc(221,36): error: use of undeclared identifier 'AOT_ContextScope_element_size'; did you mean 'ContextScope_element_size'?
  COMMON_OFFSETS_LIST(CHECK_FIELD, CHECK_ARRAY, CHECK_SIZEOF,
                                   ^
../../third_party/dart/runtime\vm/compiler/runtime_offsets_extracted.h(5766,47): note: 'ContextScope_element_size' declared here
static constexpr dart::compiler::target::word ContextScope_element_size = 32;
                                              ^
../../third_party/dart/runtime/vm/dart.cc(221,36): error: use of undeclared identifier 'AOT_ContextScope_element_size'; did you mean 'ContextScope_element_size'?
  COMMON_OFFSETS_LIST(CHECK_FIELD, CHECK_ARRAY, CHECK_SIZEOF,
                                   ^
../../third_party/dart/runtime\vm/compiler/runtime_offsets_extracted.h(5766,47): note: 'ContextScope_element_size' declared here
static constexpr dart::compiler::target::word ContextScope_element_size = 32;
                                              ^
../../third_party/dart/runtime/vm/dart.cc(221,36): error: use of undeclared identifier 'AOT_ExceptionHandlers_elements_start_offset'; did you mean 'ExceptionHandlers_elements_start_offset'?
  COMMON_OFFSETS_LIST(CHECK_FIELD, CHECK_ARRAY, CHECK_SIZEOF,
                                   ^
../../third_party/dart/runtime\vm/compiler/runtime_offsets_extracted.h(5768,5): note: 'ExceptionHandlers_elements_start_offset' declared here
    ExceptionHandlers_elements_start_offset = 12;
    ^
../../third_party/dart/runtime/vm/dart.cc(221,36): error: use of undeclared identifier 'AOT_ExceptionHandlers_elements_start_offset'; did you mean 'ExceptionHandlers_elements_start_offset'?
  COMMON_OFFSETS_LIST(CHECK_FIELD, CHECK_ARRAY, CHECK_SIZEOF,
                                   ^
../../third_party/dart/runtime\vm/compiler/runtime_offsets_extracted.h(5768,5): note: 'ExceptionHandlers_elements_start_offset' declared here
    ExceptionHandlers_elements_start_offset = 12;
    ^
../../third_party/dart/runtime/vm/dart.cc(221,36): error: use of undeclared identifier 'AOT_ExceptionHandlers_element_size'; did you mean 'ExceptionHandlers_element_size'?
  COMMON_OFFSETS_LIST(CHECK_FIELD, CHECK_ARRAY, CHECK_SIZEOF,
                                   ^
../../third_party/dart/runtime\vm/compiler/runtime_offsets_extracted.h(5769,47): note: 'ExceptionHandlers_element_size' declared here
static constexpr dart::compiler::target::word ExceptionHandlers_element_size =
                                              ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
[96/872] CXX obj/third_party/dart/runtime/vm/libdart_vm_precompiled_runtime_product.cpu_ia32.obj
FAILED: obj/third_party/dart/runtime/vm/libdart_vm_precompiled_runtime_product.cpu_ia32.obj
ninja -t msvc -e environment.x86 -- ../../buildtools/windows-x64/clang/bin/clang-cl.exe -m32 /nologo /showIncludes @obj/third_party/dart/runtime/vm/libdart_vm_precompiled_runtime_product.cpu_ia32.obj.rsp /c ../../third_party/dart/runtime/vm/cpu_ia32.cc /Foobj/third_party/dart/runtime/vm/libdart_vm_precompiled_runtime_product.cpu_ia32.obj /Fdobj/third_party/dart/runtime/vm/libdart_vm_precompiled_runtime_product_cc.pdb
In file included from ../../third_party/dart/runtime/vm/cpu_ia32.cc:11:
../../third_party/dart/runtime\vm/compiler/assembler/assembler.h(9,2): error: "AOT runtime should not use compiler sources (including header files)"
#error "AOT runtime should not use compiler sources (including header files)"
 ^
In file included from ../../third_party/dart/runtime/vm/cpu_ia32.cc:11:
In file included from ../../third_party/dart/runtime\vm/compiler/assembler/assembler.h:21:
../../third_party/dart/runtime\vm/compiler/assembler/assembler_ia32.h(9,2): error: "AOT runtime should not use compiler sources (including header files)"
#error "AOT runtime should not use compiler sources (including header files)"
 ^
In file included from ../../third_party/dart/runtime/vm/cpu_ia32.cc:11:
In file included from ../../third_party/dart/runtime\vm/compiler/assembler/assembler.h:21:
In file included from ../../third_party/dart/runtime\vm/compiler/assembler/assembler_ia32.h:18:
../../third_party/dart/runtime\vm/compiler/assembler/assembler_base.h(9,2): error: "AOT runtime should not use compiler sources (including header files)"
#error "AOT runtime should not use compiler sources (including header files)"
 ^
3 errors generated.
[107/872] CXX obj/third_party/dart/runtime/vm/libdart_vm_precompiled_runtime_product.object_graph_copy.obj
ninja: build stopped: subcommand failed.
```

Reply:
Nice & Thanks! Should I wait this dart commit rolled into engine? 
https://github.com/dart-lang/sdk/commit/2347481619f96a9263394b42697b563081f941cc

Reply:
Yeah, it would be better to wait so that you can test this PR locally before landing.